packages/aws-cdk-lib/aws-iam/lib/oidc-provider2.ts

```diff
@@ -0,0 +1,168 @@
+   * provider. For example, assume that the OIDC provider is server.example.com
+   * and the provider stores its keys at
+   * https://keys.server.example.com/openid-connect. In that case, the
+   * thumbprint string would be the hex-encoded SHA-1 hash value of the
+   * certificate used by https://keys.server.example.com.
+   *
+   * Obtain the thumbprint of the root certificate authority from the provider's
+   * server as described in https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_providers_create_oidc_verify-thumbprint.html
+   */
+  readonly thumbprints: string[];
+}
+
+/**
+ * IAM OIDC identity providers are entities in IAM that describe an external
+ * identity provider (IdP) service that supports the OpenID Connect (OIDC)
+ * standard, such as Google or Salesforce. You use an IAM OIDC identity provider
+ * when you want to establish trust between an OIDC-compatible IdP and your AWS
+ * account. This is useful when creating a mobile app or web application that
+ * requires access to AWS resources, but you don't want to create custom sign-in
+ * code or manage your own user identities.
+ *
+ * @see http://openid.net/connect
+ * @see https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_providers_oidc.html
+ *
+ * @resource AWS::IAM::OIDCProvider
+ */
+export class OpenIdConnectProvider2
+  extends Resource
+  implements IOpenIdConnectProvider2 {
+  /**
+   * Imports an Open ID connect provider from an ARN.
+   * @param scope The definition scope
+   * @param id ID of the construct
+   * @param openIdConnectProviderArn the ARN to import
+   */
+  public static fromOpenIdConnectProviderArn(
+    scope: Construct,
+    id: string,
+    openIdConnectProviderArn: string,
+  ): IOpenIdConnectProvider2 {
+    const resourceName = Arn.extractResourceName(
+      openIdConnectProviderArn,
+      'oidc-provider',
+    );
+
+    class Import extends Resource implements IOpenIdConnectProvider2 {
+      public readonly openIdConnectProviderArn = openIdConnectProviderArn;
+      public readonly openIdConnectProviderIssuer = resourceName;
+    }
+
+    return new Import(scope, id);
+  }
+
+  /**
+   * The Amazon Resource Name (ARN) of the IAM OpenID Connect provider.
+   */
+  public readonly openIdConnectProviderArn: string;
+
+  public readonly openIdConnectProviderIssuer: string;
+
+  /**
+   * The thumbprints configured for this provider.
+   */
+  public readonly openIdConnectProviderthumbprints: string;
+
+  /**
+   * Defines an OpenID Connect provider.
+   * @param scope The definition scope
+   * @param id Construct ID
+   * @param props Initialization properties
+   */
+  public constructor(
+    scope: Construct,
+    id: string,
+    props: OpenIdConnectProvider2Props,
+  ) {
+    super(scope, id, {
+      physicalName: props.oidcProviderName,
+    });
+
+    const resource = new CfnOIDCProvider(this, 'Resource', {
+      url: props.url,
+      clientIdList: props.clientIds,
+      thumbprintList: props.thumbprints,
+    });
```

Comment:
The [implementation](https://github.com/aws/aws-cdk/blob/fcb7231d2239b12c027e573f07d8d01ff357035e/packages/aws-cdk-lib/aws-iam/lib/oidc-provider.ts#L150-L153) for the original `OpenIdConnectProvider` passed in a `CodeHash` from the provider so that CFN invokes the UPDATE handler when there are code change but the properties of the resource haven't changed. 

Is this problem is fixed by using `CfnOIDCProvider`?

For more context: https://github.com/aws/aws-cdk/pull/22802/files#r1018838729

Reply:
thank you for the comment. I shall look into this